FBKVOController/FBKVOController.m

```diff
@@ -275,12 +288,20 @@ - (void)unobserve:(id)object info:(_FBKVOInfo *)info
   if (nil == info) {
     return;
   }
-  
+
+  if (!info->_registered) {
+      // this could happen when `NSKeyValueObservingOptionInitial` is one of the NSKeyValueObservingOptions.
+      // we cannot unobserve them right now because they are not technially observers yet.
+      // instead, we set a flag so they can be unobserved later, after the `addObserver` call returns.
+      info->_unregisteredBeforeRegistered = YES;
+      return;
```

Comment:
Bailing out early here will actually never remove the observation info from this context - instead you probably would want to check the above mentioned state and guard the actual `removeObserver:forKeyPath:context:` which is the source of the problem.